scripts/download_osm_data.py

```diff
@@ -96,7 +129,12 @@ def convert_iso_to_geofk(
     out_path = Path.joinpath(store_path_resources, "out")
     out_formats = ["csv", "geojson"]
     new_files = os.listdir(out_path)
-
+    microgrid_coordinates = snakemake.config["microgrids_list"]["microgrid_1"]
```

Comment:
This currently works for a microgrid only.
We should loop over all microgrids in microgrids_list to download the data for each microgrid independently and then put them together.
We can do this in two ways:
- repeat the process, create a file for each microgrid and then merge all the files
- make only a single overpass query and update the query "overpass_query" to account for the different microgrids. This basically means to add into the query the logic of searching in the box of the first microgrid or the box of the second microgrid and so on and so forth. I'm unsure this is possible but we can investigate

Reply:
secondly, ideally, we should load the microgrid_list using params, as:
microgrids_list = snakemake.params.microgrids_list

beyond this, in the snakefile, under download_osm_data, we should add:
```
params:
    ...
    microgrids_list = config["microgrids_list"]
```